pex/pex_info.py

```diff
@@ -143,9 +151,7 @@ def __init__(self, info=None):
             raise ValueError(
                 "PexInfo can only be seeded with a dict, got: " "%s of type %s" % (info, type(info))
             )
-        self._pex_info = info or {}
-        if "inherit_path" in self._pex_info:
```

Comment:
N.B.: This dance was just enforcing coercion of old bool values into new str enum values. This is now centralized in `InheritPath.for_value` so there is no longer a need here.